File: aluguel-api/src/main/kotlin/com/nortetec/imovel/ImovelController.java
package com.nortetec.imovel;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("imovel")
public class ImovelController {
    private final ImovelService service;

    public ImovelController(ImovelService service) {
        this.service = service;
    }

    @GetMapping
    public List<ImovelImpl> listaImoveis() {
        return service.listaImovel();
    }
}

File: aluguel-api/src/main/kotlin/com/nortetec/imovel/ImovelController.kt
package com.nortetec.imovel

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("imovel")
public class ImovelController(val service: ImovelService) {
    @GetMapping
    fun listaImoveis(): List<Imovel> {
        return service.listaImovel()
    }
}